theano/tensor/signal/downsample.py

```diff
@@ -635,44 +634,70 @@ def c_code(self, node, name, inp, out, sub):
           Py_XDECREF(%(gx)s);
           %(gx)s = (PyArrayObject*) PyArray_ZEROS(4, PyArray_DIMS(%(x)s), x_typenum,0);
         }
```

Comment:
What's the C code I need to write to explicitly set an already existed gx
to all zero?

On Fri, May 1, 2015 at 11:09 AM, Frédéric Bastien <notifications@github.com>
wrote:

> In theano/tensor/signal/downsample.py
> <https://github.com/Theano/Theano/pull/2843#discussion_r29506855>:
>
> > @@ -635,44 +634,70 @@ def c_code(self, node, name, inp, out, sub):
> >            Py_XDECREF(%(gx)s);
> >            %(gx)s = (PyArrayObject*) PyArray_ZEROS(4, PyArray_DIMS(%(x)s), x_typenum,0);
> >          }
>
> I mean, you need to add an else clause to this if. In it, you init to 0.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2843/files#r29506855>.
>